### ISSUE TITLE
ansible-lint not processing include with relative directory same as ansible

### ISSUE THREAD

**Issue by adelosa** [ansible/ansible-lint#160]
# Issue Type
- Bug report
# Ansible and Ansible Lint details

```
ansible 2.1.0.0
ansible-lint 2.7.1
```
- ansible installation method: pip
- ansible-lint installation method: pip
# Desired Behaviour

`ansible-lint` is able to parse relative include statements within roles the same as `ansible`
# Actual Behaviour (Bug report only)

`ansible-lint` uses role directory as base when following an include statement.
`ansible` uses the playbook directory as base when following an include statement
## playbook

{{playbook_dir}}/test-include.yml

``` YAML

---
- hosts: localhost
  connection: local
  roles:
    - test-role
```

{{playbook_dir}}/roles/test-role/tasks/main.yml

``` YAML

---
   - name: Include playbook from another role
     include: roles/include-role/tasks/main.yml
```

{{playbook_dir}}/roles/include-role/tasks/main.yml

``` YAML

---
   - debug: msg="Test including an include role"
```
## output from ansible

```
PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [127.0.0.1]

TASK [test-role : Include playbook from another role] **************************
included: /apps/dev/playbooks/ant/lint-include/roles/include-role/tasks/main.yml for 127.0.0.1

TASK [test-role : debug] *******************************************************
ok: [127.0.0.1] => {
    "msg": "Test including an include role"
}

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=3    changed=0    unreachable=0    failed=0
```
## output of running ansible-lint (test-include.yml)

```
WARNING: Couldn't open /apps/dev/playbooks/ant/lint-include/roles/test-role/tasks/roles/include-role/tasks/main.yml - No such file or directory
```

**Comment by adelosa** [comment]
Just tested adding the {{ playbook_dir }} to the include to stop making it relative.. Same result.
{{playbook_dir}}/roles/test-role/tasks/main.yml
```YAML
---
   - name: Include playbook from another role
     include: "{{ playbook_dir }}/roles/include-role/tasks/main.yml"
```

**Comment by tetraf** [comment]
Same thing with "global" tasks included in roles.
If, in "{{ playbook_dir }}/roles/foo/tasks/main.yml", I include bar.yml, Ansible seeks inside "{{ playbook_dir }}/tasks/" if not found in "{{ playbook_dir }}/roles/foo/tasks/", but ansible-lint not.

Works in ansible but not in ansible-lint:
```
$ tree
├── roles
│   └── foo
│       └── tasks
│           └── main.yml
└── tasks
    └── bar.yml
```
```
---
# File: roles/foo/tasks/main.yml
- include: bar.yml
```

**Comment by ssbarnea** [comment]
Any plans to address this?

**Comment by gd-asharov** [comment]
+1

**Comment by willthames** [comment]
If people find a fix for this, with passing tests, I'll look at incorporating it.

**Comment by marcjay** [comment]
I don't suppose the last update to this has changed now that ansible-lint is part of Ansible proper?

**Comment by greenflowers** [comment]
Where would I start if I want to implement that?

**Comment by zeppelinen** [comment]
I am using Ansible patched with this PR: https://github.com/ansible/ansible/pull/58725

I achieved to make it work with the following syntax
```
    - name: Load environment
      include_tasks: tasks/load_environment.yml
      args:
        tags:
          - always
```

Please note it will always fail to find if using `file` key.

The whole problem is related to https://github.com/ansible/ansible-lint/issues/507 that is still not fixed for 2.8 version.
